lib/instrumentation/transaction.js

```diff
@@ -93,7 +93,26 @@ Transaction.prototype.startSpan = function (name, type) {
   }
   this._builtSpans++
 
-  return new Span(this, name, type)
+  if (!opts) {
+    opts = {
+      timer: this._timer,
+      traceContext: (this._agent._instrumentation.activeSpan || this)._context
+    }
+  } else if (typeof opts === 'string') {
+    // To be backwards compatible with the old API, we also accept the
+    // traceparent as a string
+    opts = {
+      timer: this._timer,
+      traceparent: opts
+    }
+  } else {
+    opts.timer = this._timer
+    if (!opts.traceparent) {
+      opts.traceContext = (this._agent._instrumentation.activeSpan || this)._context
+    }
+  }
```

Comment:
Maybe move this logic _into_ the `Span` constructor? It still has access to the transaction there, and that would mean the tests don't need to explicitly pass the `traceContext` in everywhere in the span test file.

Reply:
Fixed